src/client/driver/driver.js

```diff
@@ -476,11 +477,51 @@ export default class Driver extends serviceUtils.EventEmitter {
             });
     }
 
+    _prepareWindowPredicateParam () {
+        const parsedUrl = hammerhead.utils.url.parseProxyUrl(window.location.toString());
+
+        const { protocol, host, port, hostname, partAfterHost: query } = parsedUrl.destResourceInfo;
+
+        const location = {
+            href: parsedUrl.destUrl,
+            protocol,
+            host,
+            port,
+            hostname,
+            query
+        };
+
+        return {
+            location,
+            title: document.title
+        };
+    }
+
+    _checkWindowExists (msg) {
+        return msg.windowId === this.windowId || this._checkWindowByPredicate(msg);
+    }
+
+    _checkWindowByPredicate ({ fn }) {
+        if (!fn)
+            return false;
+
+        const executor = new ClientFunctionExecutor(fn);
+
+        try {
+            const windowObj = this._prepareWindowPredicateParam();
+
+            return executor.fn.call(null, windowObj);
+        }
+        catch (err) {
+            return false;
```

Comment:
Need to notify a customer about errors during predicate execution.
For example customer specifies the predicate that raises an JavaScript error.
```js
await t.switchToWindow(w => {  return w.locaion.href === '/test' }) // typo in the the 'location' property name.
```